smart-common/src/main/java/org/smartdata/BloomFilter.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.List;
+
+public class BloomFilter {
```

Comment:
Here is an open lib implemented by google ([guava bloom filter](https://github.com/google/guava/blob/master/guava/src/com/google/common/hash/BloomFilter.java)), we can refine our method based on it.